packages/flutter/lib/src/material/toggleable.dart

```diff
@@ -225,14 +229,12 @@ abstract class RenderToggleable extends RenderConstrainedBox {
       ..curve = Curves.easeIn
       ..reverseCurve = Curves.easeOut;
     if (tristate) {
-      switch (_positionController.status) {
-        case AnimationStatus.forward:
-        case AnimationStatus.completed:
-          _positionController.reverse();
-          break;
-        default:
-          _positionController.forward();
-      }
+      if (value == null)
+        _positionController.value = 0.0;
+      if (value != false)
+        _positionController.forward();
+      else
+        _positionController.reverse();
```

Comment:
What is this change for?

Reply:
The `_positionController` has the value 0.0 when a toggleable is inactive and it goes to 1.0 when it is active (this is from the docs of the variable). It is used in the animations.

However, for a tristate Checkbox, this was not followed. Instead if went from 1.0 to 0.0 and 0.0 to 1.0 depending on how many times you had tapped the Checkbox. I made this change to make sure it will go to 1.0 when the tristate Checkbox is selected/active.

I needed to do this to make sure that the color of the overlay is correct, as I am using the `_position.value` to lerp between the inactive and active color:
https://github.com/flutter/flutter/pull/71569/files#diff-4f3e3ab7fe1911bbf70b937e5692e96f25acbdd789da30d3e9e993d6113e2210R485